Mage.Sets/src/mage/sets/darksteel/PulseOfTheTangle.java

```diff
@@ -0,0 +1,117 @@
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Rarity;
+import mage.constants.Zone;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.permanent.token.BeastToken;
+import mage.players.Player;
+import mage.target.TargetPlayer;
+import mage.target.common.TargetOpponent;
+
+/**
+ *
+ * @author wetterlicht
+ */
+public class PulseOfTheTangle extends CardImpl {
+
+    public PulseOfTheTangle(UUID ownerId) {
+        super(ownerId, 80, "Pulse of the Tangle", Rarity.RARE, new CardType[]{CardType.SORCERY}, "{1}{G}{G}");
+        this.expansionSetCode = "DST";
+
+        // Put a 3/3 green Beast creature token onto the battlefield. Then if an opponent controls more creatures than you, return Pulse of the Tangle to its owner's hand.
+        this.getSpellAbility().addEffect(new CreateTokenEffect(new BeastToken()));
+        this.getSpellAbility().addEffect(new PulseOfTheTangleReturnToHandEffect());
+    }
+
+    public PulseOfTheTangle(final PulseOfTheTangle card) {
+        super(card);
+    }
+
+    @Override
+    public PulseOfTheTangle copy() {
+        return new PulseOfTheTangle(this);
+    }
+}
+
+class PulseOfTheTangleReturnToHandEffect extends OneShotEffect {
+
+    PulseOfTheTangleReturnToHandEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "Then if an opponent controls more creatures than you, return Pulse of the Tangle to its owner's hand";
+    }
+
+    PulseOfTheTangleReturnToHandEffect(final PulseOfTheTangleReturnToHandEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PulseOfTheTangleReturnToHandEffect copy() {
+        return new PulseOfTheTangleReturnToHandEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        FilterControlledCreaturePermanent controllerFilter = new FilterControlledCreaturePermanent();
+        PermanentsOnBattlefieldCount controllerCount = new PermanentsOnBattlefieldCount(controllerFilter);
+        boolean check = false;
+        if (controller != null) {
+            for (UUID opponentID : game.getOpponents(controller.getId())) {
+                if (opponentID != null) {
+                    FilterCreaturePermanent opponentFilter = new FilterCreaturePermanent();
+                    opponentFilter.add(new ControllerIdPredicate(opponentID));
+                    PermanentsOnBattlefieldCount opponentCreatureCount = new PermanentsOnBattlefieldCount(opponentFilter);
+                    check = opponentCreatureCount.calculate(game, source, this) > controllerCount.calculate(game, source, this);

```

Comment:
it would be a bit better to only calculate the controllers number once before the loop